aws/resource_aws_iam_role.go

```diff
@@ -120,6 +124,34 @@ func resourceAwsIamRole() *schema.Resource {
 				Default:      3600,
 				ValidateFunc: validation.IntBetween(3600, 43200),
 			},
+
+			"managed_policy_arns": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
+			"inline_policies": {
+				Type:             schema.TypeMap,
```

Comment:
If the provider SDK is acting really strange with `TypeMap` and the handling of the JSON in the value, we should likely treat this like everywhere else and go `TypeSet` here with a nested schema of optional `name` and required `policy` (with the normal JSON handlers) to not introduce a new pattern or complexity.

Reply:
Moving this direction would allow us to avoid the use of the "sidecar" attributes. I would still use the attribute internally but it would be `computed` only.

Reply:
All sidecar attributes have been removed. Was actually easier to do it without once I had paradigm shift in thinking.